examples/docker/kafka-oauth-strimzi/kafka/simple_kafka_config.sh

```diff
@@ -19,14 +19,32 @@ pop_value() {
   unset props[$key]
 }
 
+#
+# This function allows you to encode as KAFKA_* env vars property names that contain characters invalid for env var names
+# You can use:
+#   KAFKA_LISTENER_NAME_CLIENT_SCRAM__2DSHA__2D256_SASL_JAAS_CONFIG=something
+#
+# Which will first be converted to:
+#   KAFKA_LISTENER_NAME_CLIENT_SCRAM%2DSHA%2D256_SASL_JAAS_CONFIG=something
+#
+# And then to:
+#   KAFKA_LISTENER_NAME_CLIENT_SCRAM-SHA-256_SASL_JAAS_CONFIG=something
+#
+unescape() {
+  if [[ "$1" != "" ]]; then
+    echo "$1" | sed 's/__/\%/g' | sed "s@+@ @g;s@%@\\\\x@g" | xargs -0 printf "%b"
```

Comment:
It's just how this solution evolved, not that it's the optimal solution, but it works. The second `sed` is something I found googling, it worked so I didn't really explore further. The first sed is separate because it has to happen in a separate pass (pssst I'm no `sed` expert :). 

I could replace it with single invocation with '-e' but then I have to test it again to make sure it works. TBH at the moment I'd prefer to work on other things - the cost-benefit analysis with this one is not great :)

Reply:
Nevermind, it was easier to test than it seemed at first.
Changed to use -e twice for a single `sed` invocation.